llvm/tools/sycl-post-link/SYCLDeviceRequirements.cpp

```diff
@@ -56,24 +67,23 @@ void llvm::getSYCLDeviceRequirements(
         std::vector<uint32_t>(Values.begin(), Values.end());
   }
 
-  std::optional<llvm::SmallVector<int64_t, 3>> ReqdWorkGroupSize;
+  std::optional<llvm::SmallVector<size_t, 3>> ReqdWorkGroupSize;
   for (const Function &F : MD.getModule()) {
     if (const MDNode *MDN = F.getMetadata("reqd_work_group_size")) {
-      llvm::SmallVector<int64_t, 3> NewReqdWorkGroupSize;
-      for (const auto &operand : MDN->operands())
+      llvm::SmallVector<size_t, 3> NewReqdWorkGroupSize;
+      for (size_t I = 0, E = MDN->getNumOperands(); I < E; ++I)
         NewReqdWorkGroupSize.push_back(
-            ExtractIntegerFromMDNodeOperand(operand));
+            ExtractUnsignedIntegerFromMDNodeOperand(MDN, I));
       if (!ReqdWorkGroupSize)
         ReqdWorkGroupSize = NewReqdWorkGroupSize;
-      else if (!std::equal(ReqdWorkGroupSize->begin(), ReqdWorkGroupSize->end(),
-                           NewReqdWorkGroupSize.begin())) {
-        // two functions in the module have different required work group sizes
-      }
+      else
+        assert(std::equal(ReqdWorkGroupSize->begin(), ReqdWorkGroupSize->end(),
+                          NewReqdWorkGroupSize.begin()));
     }
   }
 
   if (ReqdWorkGroupSize)
-    Requirements["reqd_work_group_size"] = std::vector<uint32_t>(
+    Requirements["reqd_work_group_size_size_t"] = std::vector<size_t>(
```

Comment:
A comment is needed here that the property name should be `reqd_work_group_size`, but we couldn't change the type of existing property because we missed ABI break window; therefore, we need to rename it back (to a more obvious name) once ABI break is allowed.